File: server/src/main/java/com/db/datastoreserver/service/dto/PostCreateRequest.java
package com.db.datastoreserver.service.dto;

import com.db.datastoreserver.domain.member.Member;
import com.db.datastoreserver.domain.post.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PostCreateRequest {
    private String title;
    private String content;
    private Location location;
    private Long price;
    private Category category;

    public Post toEntity(Member member, List<Photo> photos) {
        return Post.builder()
                .title(title)
                .author(member)
                .content(content)
                .location(location)
                .category(category)
                .price(price)
                .status(Status.SALE)
                .photos(photos)
                .build();
    }
}

File: k-datastore-server/src/main/kotlin/com/db/kdatastoreserver/service/PostCreateRequest.kt
package com.db.kdatastoreserver.service

import com.db.kdatastoreserver.domain.Category
import org.springframework.web.multipart.MultipartFile

data class PostCreateRequest(
    val title: String,
    val content: String,
    val photos: List<MultipartFile?> = emptyList(),
    val longitude: String,
    val latitude: String,
    val price: Int,
    val category: Category,
)
